### PR TITLE
Fix 149, commit append to the last line

### PR DESCRIPTION
**Pull request by iaia**
ref. #149 

When set commit at the end of default section, append to the last line.

### CODE REVIEW

**Comment by jreybert** [comment]
Just to let you know, I watched your PR the first day, but I did not found time to test it.

Seems quite good to me. I just wonder the `w$` pattern, maybe the pattern `end of file`would be better (i can't remember right now the exact syntax of this pattern).

Thanks for the PR anyway!

**Comment by iaia** [comment]
Thank you for reviewing.
What you are saying is maybe `$EOF`? I fixed. please review again.

**Comment by jreybert** [comment]
End of file vim pattern is `\%$` (see `:help %$`). `$EOF` doe snot exists in vim.

File: plugin/magit.vim [hunk 1]
```diff
@@ -293,7 +293,13 @@ endfunction
 function! s:mg_get_commit_msg(...)
 	let commit_section_pat_start='^'.g:magit_sections.commit.'$'
 	" Get next section pattern with g:magit_default_sections order
-	let commit_section_pat_end='^'.g:magit_sections[g:magit_default_sections[match(g:magit_default_sections, 'commit')+1]].'$'
+	if ( match(g:magit_default_sections, 'commit')+1 >= len(g:magit_default_sections) )
```
Comment (jreybert) [inline plugin/magit.vim]: use intermediate variable to store `match(g:magit_default_sections, 'commit')` result
check if this position is equal to len + 1

File: plugin/magit.vim [hunk 2]
```diff
@@ -293,7 +293,13 @@ endfunction
...
+		let commit_position = line("$EOF")
```
Comment (jreybert) [inline plugin/magit.vim]: do not use intermediate variable for this, store directly in commit_section_pat_end

use `\%$` pattern, without `line()`

File: plugin/magit.vim [hunk 3]
```diff
@@ -293,7 +293,13 @@ endfunction
...
+	else
+		let commit_position = g:magit_sections[g:magit_default_sections[match(g:magit_default_sections, 'commit')+1]]
```
Comment (jreybert) [inline plugin/magit.vim]: use intermediate variable declared before

keep `^` and `$` here

**Review by jreybert — CHANGES_REQUESTED** [review]
you spotted the right codelines to fix the bug, but the fix is wrong.

The usage of `line()` makes no sense here, you don't want to check against the content of the last line (which is certainly `empty line`), because you could have empty line in your commit message. Instead, use the pattern `\%$`, which is regex pattern for end of file.

Others comments are mainly cosmetics, but I would appreciate them too.

**Comment by iaia** [comment]
@jreybert sorry, i am not good at writing vimscript.
I rewrote it. please review again. thank you.

File: plugin/magit.vim [hunk 4]
```diff
@@ -293,7 +293,13 @@ endfunction
 function! s:mg_get_commit_msg(...)
 	let commit_section_pat_start='^'.g:magit_sections.commit.'$'
 	" Get next section pattern with g:magit_default_sections order
-	let commit_section_pat_end='^'.g:magit_sections[g:magit_default_sections[match(g:magit_default_sections, 'commit')+1]].'$'
+	let commit_position = match(g:magit_default_sections, 'commit')
+	if ( commit_position + 1 == len(g:magit_default_sections) )
+	  let commit_section_pat_end='\%$'
+	else
+	  let commit_section_pat_end='^'.g:magit_sections[g:magit_default_sections[match(g:magit_default_sections, 'commit')+1]].'$'
```
Comment (jreybert) [inline plugin/magit.vim]: Reuse `commit_position` here.

**Review by jreybert — CHANGES_REQUESTED** [review]
Excellent! Just one last thing and you're good.

File: plugin/magit.vim (hunk 4 above)
Comment (iaia) [reply plugin/magit.vim]: I fixed!

**Review by jreybert — CHANGES_REQUESTED** [review]
Perfect.

I am sorry, two last things I did not noticed the first time:
* remove the file `.development.vim` from your commit
* rename the commit title to something like: "Handle commit set at the end of magit_default_sections", and add "ref #149" (the sign # is important) in the commit body.

**Comment by iaia** [comment]
oops! sorry, i deleted `.development.vim`. and change commit message.

**Review by jreybert — APPROVED** [review]

**Comment by jreybert** [comment]
Excellent, thanks for the PR and for all your changes through the review process!

**Comment by iaia** [comment]
Thanks for the merge!

**Pull request merged**